packages/components/src/templates/next/layouts/Collection/utils/sortCollectionItems.ts

```diff
@@ -0,0 +1,96 @@
+import type { GetCollectionItemsProps } from "./getCollectionItems"
+import type { AllCardProps } from "~/interfaces"
+
+export type SortableCardProps = AllCardProps & {
+  rawDate?: Date
+}
+
+export interface SortCollectionItemsProps
+  extends Pick<GetCollectionItemsProps, "sortBy" | "sortDirection"> {
+  items: SortableCardProps[]
+}
+
+// Sort by last updated date, tiebreaker by title
+const sortCollectionItemsByDate = ({
+  items,
+  sortDirection = "desc",
+}: Omit<SortCollectionItemsProps, "sortBy">) => {
+  return items.sort((a, b) => {
+    // If both items have no dates, sort by title ascending as a tiebreaker
+    if (a.rawDate === undefined && b.rawDate === undefined) {
+      return a.title.localeCompare(b.title)
+    }
+
+    // Rank items with no dates last
+    if (a.rawDate === undefined) {
+      return 1
+    }
+
+    if (b.rawDate === undefined) {
+      return -1
+    }
+
+    // If both items have same date, sort by title ascending as a tiebreaker
+    if (a.rawDate.getTime() === b.rawDate.getTime()) {
+      return a.title.localeCompare(b.title)
+    }
+
+    switch (sortDirection) {
+      case "asc":
+        return a.rawDate.getTime() < b.rawDate.getTime() ? 1 : -1
+      case "desc":
+        return a.rawDate.getTime() > b.rawDate.getTime() ? 1 : -1
+      default:
+        const exhaustiveCheck: never = sortDirection
+        return exhaustiveCheck
+    }
+  }) as AllCardProps[]
+}
+
+// Sort by title, tiebreaker by last updated date
+const sortCollectionItemsByTitle = ({
+  items,
+  sortDirection = "asc",
+}: Omit<SortCollectionItemsProps, "sortBy">) => {
+  return items.sort((a, b) => {
+    // Rank items with no dates last
```

Comment:
why are we ranking by date first for a sort by title logic?

the date should only be a tiebreaker when the titles are equal right?

Reply:
i.e. 2 conditional flows:

if titles are different -> localeCompare

if titles are same -> use date as tiebreaker

if both have no dates -> consider as equal, else rank item with no date as last

Reply:
hmm i don't think it matter since they are mutually exclusive case (just happy-path first v.s. sad-path guard-clausish)?

nevertheless have updated the code to do happy path first since it reads better logically + added variable for conditions to improve readability too! https://github.com/opengovsg/isomer/pull/1212/commits/c819ab6cce71e4664237fb97fafa21afc09269f8